agent/structs/config_entry_mesh.go

```diff
@@ -20,6 +20,14 @@ type MeshConfigEntry struct {
 	// MutualTLSMode=permissive in either service-defaults or proxy-defaults.
 	AllowEnablingPermissiveMutualTLS bool `json:",omitempty" alias:"allow_enabling_permissive_mutual_tls"`
 
```

Comment:
Do we need non-api folder changes? Were pipelines not working without this?

Reply:
It makes sense to me that api and proto folder are included since those are modules that we publish, but do we need the agent/structs and website/content changes?

Reply:
proto/private/pbconfigentry/config_entry.gen.go has `func MeshConfigToStructs(s *MeshConfig, t *structs.MeshConfigEntry)` with logic that requires the struct has `ValidateClusters`

https://github.com/hashicorp/consul/pull/21634/files#diff-5d7ac9600744c7e7aca17405abe5a790ecf073be38959a35576edc66f005220cR1692-R1727